tests/vanilla/store.test.tsx

```diff
@@ -933,3 +933,33 @@ it('should call subscribers after setAtom updates atom value on mount but not on
   expect(unmount).toHaveBeenCalledTimes(1)
   expect(listener).toHaveBeenCalledTimes(0)
 })
+
+it('processes deep atom a graph beyond maxDepth', () => {
+  function getMaxDepth() {
+    let depth = 0
+    function d(): number {
+      ++depth
+      try {
+        return d()
+      } catch (error) {
+        return depth
+      }
+    }
+    return d()
+  }
+  const maxDepth = getMaxDepth()
```

Comment:
Not sure this is ideal in the test, but it looks like a nice hack in the meantime.